ibis/tests/all/test_window.py

```diff
@@ -15,6 +16,63 @@
 )
 
 
+def ntile(x, bucket: int):
+    """
+    NTILE divides given data set into a number of buckets.
+
+    It divides an ordered and grouped data set into a number of buckets
+    and assigns the appropriate bucket number to each row.
+
+    Return an integer ranging from 0 to `bucket - 1`, dividing the
+    partition as equally as possible.
+
+    Parameters
+    ----------
+    x : pandas.core.groupby.generic.SeriesGroupBy ||
+        pd.core.groupby.generic.DataFrameGroupBy ||
+        pandas.Series || pandas.DataFrame
+    bucket: int
+
+    Returns
+    -------
+    pandas.Series
+    """
+    # internal ntile function
+    def _ntile(x: pd.Series, bucket: int):
+        n = x.shape[0]
+        sub_n = n // bucket
+        diff = n - (sub_n * bucket)
+
+        result = []
+        for i in range(bucket):
+            sub_result = [i] * (sub_n + (1 if diff else 0))
+            result.extend(sub_result)
+            if diff > 0:
+                diff -= 1
+        return pd.Series(result, index=x.index)
```

Comment:
why are you evaluating directly in the api like this? 

are the ntile ops not defined?

these should be dispatched generically, not done like this.

Reply:
pandas doesn't have `ntile` operation.
the most similar operation is`qcut` but it seems there are some differences.

what is your recommendation in this case?